src/modules/Metadata/Item/index.ts

```diff
@@ -0,0 +1,57 @@
+import { ItemWearableMetadata } from '../../../entities/schema'
+import { toLowerCase } from '../../../utils'
+import { Categories, BodyShapes } from './constants'
+
+export function buildWearableMetadata(
+  id: string,
+  metadataValues: string[]
+): ItemWearableMetadata | null {
+  if (metadataValues.length < 1) {
+    return null
+  }
+
+  if (metadataValues[1] == '1') {
+    return buildWearableMetadataV1(id, metadataValues)
+  }
+
+  return null
+}
+
+function buildWearableMetadataV1(
+  id: string,
+  metadataValues: string[]
+): ItemWearableMetadata | null {
+  if (metadataValues.length !== 6) {
+    return null
+  }
+
+  let itemMetadata = ItemWearableMetadata.load(id)
+  if (itemMetadata == null) {
+    itemMetadata = new ItemWearableMetadata(id)
+  }
+
+  itemMetadata.name = metadataValues[2]
+  itemMetadata.description = metadataValues[3]
+  itemMetadata.category = buildWearableCategory(metadataValues[4])
+  itemMetadata.bodyShapes = buildWearableBodyShapes(metadataValues[5])
+
+  itemMetadata.save()
+
+  return itemMetadata
+}
+
+function buildWearableCategory(metadataCategory: string): string | null {
+  let lowerCasedMetadataCategory = toLowerCase(metadataCategory)
+  if (Categories.indexOf(lowerCasedMetadataCategory) !== 1) {
```

Comment:
Are you sure that this works?

Reply:
Are we using the same version of assemblescrypt that the one that supports `indexOf` ?

Reply:
I've updated the version of @graphprotocol/graph-ts to 0.22.1 and that version uses the version 0.19.10 of AssemblyScript that contains the [indexOf](https://github.com/AssemblyScript/assemblyscript/blob/b7a3db8c533584f438695fd58565cc2effd7f898/std/assembly/array.ts#L193) operation here.